### PR TITLE
asserts: introduce device assertion

### PR DESCRIPTION
**Pull request by pedronis**
This introduces the device assertion. 

In order to support putting the device key in the headers readably I added simple support for header values with newlines in them, they just get 1-space indented which is a reversible transformation, added test for this. 

This extension uses a fairly strict format, it could be relaxed in a backward compatible way by using one line look-ahead and some amount of space trimming if the flexibility on the creation side seems later useful.

### CODE REVIEW

**Comment by pedronis** [comment]
requires #460 

I stopped short of adding checkConsistency for this one, because it seems that approach needs some reviewing, like it would make sense to cross reference there the model assertion but that may imply knowing the series we operate as.

File: asserts/device_asserts.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+// Timestamp returns the time when the device assertion was issued.
+func (dev *Device) Timestamp() time.Time {
+	return dev.timestamp
+}
+
+// TODO: implement further consistency checks for Device but first review approach
+
+func assembleDevice(assert assertionBase) (Assertion, error) {
+	// TODO: authority-id can only == canonical or brand-id
```
Comment (emgee) [inline asserts/device_asserts.go]: This feels like a check the client should make.

**Comment by emgee** [comment]
The device assertions looks fine. However, I have some concerns about using a significant <space>, e.g. `name:` vs `name: `, to detect simple vs multi-line. I guess it depends if the goal of the assertions format is only to be human *readable* and not expect a human to ever edit one (perhaps as a signature-less version).

**Comment by pedronis** [comment]
@emgee I don't understand the details of your comment,  the difference is "name: value" vs "name:\n value" for multiline, yes, is sort of  subtle, mostly I didn't want to pick further punctuation, but the library isn't oriented about human editing of assertions at all

**Comment by emgee** [comment]
If we say the assertion format is machine oriented then I'm +1 on this branch.

**Comment by pedronis** [comment]
the code could also do one line look ahead, but then the question is really about accepting arbitrary spacing after ':' and arbitrary consistent indent I suppose because worrying about/accepting arbitrary just for multiline would be weird

**Comment by pedronis** [comment]
also we should be careful to not make the space of assertion texts that mean the same thing too large

File: asserts/asserts.go [hunk 2]
```diff
@@ -158,16 +160,54 @@ func parseHeaders(head []byte) (map[string]string, error) {
...
-	for _, entry := range strings.Split(string(head), "\n") {
-		nameValueSplit := strings.Index(entry, ": ")
+	lines := strings.Split(string(head), "\n")
+	for i := 0; i < len(lines); {
+		entry := lines[i]
+		i++
+		nameValueSplit := strings.Index(entry, ":")
 		if nameValueSplit == -1 {
-			return nil, fmt.Errorf("header entry missing name value ': ' separation: %q", entry)
+			return nil, fmt.Errorf("header entry missing name/value ':' separator: %q", entry)
```
Comment (chipaca) [inline asserts/asserts.go]: Much as I enjoy garden path sentences, I don't think they should be used for error messages. Perhaps a better option would be just

> header entry missing ':' separator

or

> header entry missing ':' separator between name and value

?

**Comment by chipaca** [comment]
Nitpick aside, :+1:

**Comment by zyga** [comment]
+1 LGTM

**Comment by emgee** [comment]
+1 lgtm

**Pull request merged**